### DISCUSSION TITLE
Betterproto plugin to link this project for gNMI fields

### DISCUSSION THREAD

**Discussion by omikader** [python-gnxi/python-gnmi-proto#16] [Ideas] [answered]
Is there a way for `betterproto` to automatically link this project when compiling a raw `.proto` file that has a gNMI field? Take this `.proto` [file](https://github.com/onosproject/onos-api/blob/master/proto/onos/config/admin/admin.proto#L100) for example. When running it through `betterproto`, the output [file](https://github.com/onosproject/onos-api/blob/master/python/onos/config/admin/__init__.py#L107) is looking for a local implementation of gNMI. My solution so far is to manually edit the file to refer to this library after autogenerating it, but I was wondering if there was a better option.

**Comment by abn** [comment, marked as answer]
I suspect this needs to be supported by `betterproto`. Not sure if we can inject custom mappings during compilation, similar to how the `google.proto` types are mapped.

**Reply by abn** [reply]
I have asked this on the [betterproto slack team](https://betterproto.slack.com/archives/C014C74KU7N/p1621991146003500).
